scripts/core/editor3/reducers/spellchecker.jsx

```diff
@@ -28,28 +24,47 @@ const spellchecker = (state = {}, action) => {
 const replaceWord = (state, text) => {
     const {editorState, spellcheckerMenu} = state;
     const {word, editorSelection} = spellcheckerMenu;
-
-    var textSelection;
-    var newContentState;
-    var newEditorState;
-
-    textSelection = editorSelection.merge({
+    const sel = editorSelection.merge({
         anchorOffset: word.offset,
         focusOffset: word.offset + word.text.length
     });
 
-    newContentState = Modifier.replaceText(editorState.getCurrentContent(), textSelection, text);
-    newEditorState = EditorState.push(editorState, newContentState, 'spellchecker');
+    var newState = Modifier.replaceText(editorState.getCurrentContent(), sel, text);
 
-    return Object.assign({}, state, {
-        editorState: newEditorState,
-        spellcheckerMenu: null
-    });
+    newState = EditorState.push(editorState, newState, 'spellchecker');
+
+    return {
+        ...state,
+        editorState: newState,
+        spellcheckerMenu: {
+            ...spellcheckerMenu,
+            visible: false
+        }
+    };
 };
 
 /**
  * @ngdoc method
- * @name Spellchecker Reducers#showContextMenu
+ * @name closeContextMenu
+ * @param {Object} state
+ * @return {Object} returns new state
+ * @description Save on store all relevant data for spellchecker context menu
+ */
+const closeContextMenu = (state) => {
```

Comment:
yep I think it's component state rather than application state, think of eg datetime picker, that's also not affecting application state imo if you open the calendar, or when using dropdown menu.
there is a span added for errors and on hover this should be displayed, but not modify application state.
that component shouldn't be part of dom if not visible and then there is no need for state to control it.